File: book-service/src/main/java/com/bookservice/model/Book.java
package com.bookservice.model;

import com.bookservice.model.enums.BookGenres;
import jakarta.annotation.Nonnull;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Nonnull
    private String title;
    @Nonnull
    private UUID authorId;
    private String description;
    @ElementCollection(targetClass = BookGenres.class)
    @Enumerated(EnumType.STRING)
    private List<BookGenres> genres;
    private LocalDate publishedDate;
    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: book-service/src/main/java/com/bookservice/model/Book.kt
package com.bookservice.model

import com.bookservice.model.enums.BookGenres
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.*

@Entity
data class Book @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,
    var title: String,
    var authorId: UUID,
    var description: String,
    @ElementCollection(targetClass = BookGenres::class)
    @Enumerated(EnumType.STRING)
    var genres: List<BookGenres>,
    var publishedDate: LocalDate,
    @CreationTimestamp
    val createdAt: LocalDateTime? = null,
    @UpdateTimestamp
    val updatedAt: LocalDateTime? = null
)